commands/project/dev/unifiedFlow.ts

```diff
@@ -167,7 +169,16 @@ export async function unifiedProjectDevFlow(
     env,
   });
 
-  await LocalDev.start();
+  await localDevProcess.start();
 
-  handleExit(({ isSIGHUP }) => LocalDev.stop(!isSIGHUP));
+  const watcher = new LocalDevWatcher(localDevProcess);
+  watcher.start();
+
+  handleKeypress(async key => {
+    if ((key.ctrl && key.name === 'c') || key.name === 'q') {
+      await Promise.all([localDevProcess.stop(), watcher.stop()]);
+    }
+  });
+
+  handleExit(({ isSIGHUP }) => localDevProcess.stop(!isSIGHUP));
```

Comment:
`LocalDevProcess` represents the current local dev session (similar to the old `LocalDevManager`). Everything that interacts with the `LocalDevProcess` now lives _outside_ of it and uses public methods to make necessary changes to the process.

This includes
* Watcher
* Keypress listener
* In the future, the websocket server

Reply:
I'd add a note with that explanation directly into the code.